doc/rtd/development/first_PR.rst

```diff
@@ -0,0 +1,171 @@
+
+If your company has signed the CLA for you, please contact us to help in
+verifying which Launchpad/GitHub accounts are associated with the company.
+
+For any questions or help with the process, email
+`James Falcon <mailto:james.falcon@canonical.com>`_ with the subject:
+"Cloud-init CLA". You can also contact user ``falcojr`` in the #cloud-init
+channel on the `Libera IRC network <IRC_>`_.
+
+Configure git
+=============
+
+Next, configure ``git`` with your email and name for commit messages.
+
+Your name will appear in commit messages and will also be used in changelogs or
+release notes. Give yourself credit! ::
+
+  git config user.name "Your Name"
+  git config user.email "Your Email"
+
+Clone the repository
+====================
+
+* Sign in to your `GitHub`_ account.
+
+* From the cloud-init `upstream repository <GH repo_>`_ on GitHub, click on the
+  :guilabel:`Fork` button.
+
+* Create a new remote pointing to your personal GitHub repository.
+
+  .. code-block:: sh
+
+      git clone git@github.com:GH_USER/cloud-init.git
+      cd cloud-init
+      git remote add upstream git@github.com:canonical/cloud-init.git
+      git push origin main
+
+  Remember to change ``GH_USER`` to your GitHub username.
+
+Work on your feature or bug
+===========================
+
+Create a new topic branch for your work: ::
+
+    git checkout -b my-topic-branch
+
+Create a virtual environment
+----------------------------
+
+It is very often helpful to create a safe and sandboxed environment to test
+your changes in while you work. If you are not sure how to do this, check out
+:ref:`our QEMU tutorial<tutorial_qemu>`, which walks through this process
+step-by-step.
+
+Make your commits
+-----------------
+
+Make and commit your changes (note, you can make multiple commits, fixes, and
+add more commits): ::
+
+    git commit
+
+Add your name to the CLA signers list
+-------------------------------------
+
+As part of your first PR to cloud-init, you should also add your GitHub
+username (alphabetically) to the in-repository list that we use to track CLA
+signatures: `tools/.github-cla-signers`_.
+
+`PR #344`_ and `PR #345`_ are good examples of what this should look like in
+your pull request.
+
+Format the code
+---------------
+
+Apply the ``black`` and ``isort`` formatting rules with `tox`_: ::
+
+    tox -e do_format
+
+Run unit tests
+--------------
+
+Run unit tests and lint/formatting checks with `tox`_: ::
+
+    tox
+
+Push your changes
+-----------------
+
+Push your changes to your personal GitHub repository: ::
+
+    git push -u origin my-topic-branch
+
+Create a pull request
+=====================
+
+Use your browser to create a pull request:
+
+- Open the branch on GitHub
+
+- You can see a web view of your repository and navigate to the branch at: ::
+
+      https://github.com/GH_USER/cloud-init/tree/my-topic-branch
+
+- Click :guilabel:`Pull Request`.
+
+- Fill out the pull request title, summarizing the change and a longer message
```

Comment:
Hrm...I thought I had included a comment here earlier.

Since we have a [PR template](https://github.com/canonical/cloud-init/blob/main/.github/PULL_REQUEST_TEMPLATE.md) that is automatically populated anytime somebody creates  a pull request, I'm not sure we need this template here too. The two have already diverged. We can instead say that when submitting a PR to fill out all pre-populated fields accordingly.

Reply:
Yeah I did wonder about that, but since this looked more like an example I didn't want to just remove it. Happy to take it out though for ongoing maintenance reasons.